zerver/webhooks/jira/view.py

```diff
@@ -227,24 +227,29 @@ def handle_created_issue_event(payload: Dict[str, Any]) -> str:
 def handle_deleted_issue_event(payload: Dict[str, Any]) -> str:
     return u"{} **deleted** {}!".format(get_issue_author(payload), get_issue_string(payload))
 
+CONTENT_TYPE = {
+    "jira:issue_created": handle_created_issue_event,
+    "jira:issue_deleted": handle_deleted_issue_event,
+    "jira:issue_updated": handle_updated_issue_event,
+    "comment_created": handle_updated_issue_event
+}
+
 @api_key_only_webhook_view("JIRA")
 @has_request_variables
 def api_jira_webhook(request: HttpRequest, user_profile: UserProfile,
                      payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
 
     event = get_event_type(payload)
+    subject = get_issue_subject(payload)
+
     if event == 'jira:issue_created':
-        subject = get_issue_subject(payload)
-        content = handle_created_issue_event(payload)
+        content = CONTENT_TYPE[event](payload)  # type: ignore # we are basically calling the funcitons
     elif event == 'jira:issue_deleted':
-        subject = get_issue_subject(payload)
-        content = handle_deleted_issue_event(payload)
+        content = CONTENT_TYPE[event](payload)  # type: ignore # we are basically calling the funcitons
     elif event == 'jira:issue_updated':
-        subject = get_issue_subject(payload)
-        content = handle_updated_issue_event(payload, user_profile)
+        content = CONTENT_TYPE[event](payload, user_profile)  # type: ignore # we are basically calling the funcitons
     elif event == 'comment_created':
-        subject = get_issue_subject(payload)
-        content = handle_updated_issue_event(payload, user_profile)
+        content = CONTENT_TYPE[event](payload, user_profile)  # type: ignore # we are basically calling the funcitons
```

Comment:
You should be able to do something like:
``` python
content_func = JIRA_CONTENT_FUNCTION_MAPPER.get(event)
# Handle cases for where event is in ignored events or has no function and then...
content = content_func(payload)
```

If we already have a mapping of events to functions, we shouldn't need those `if` statements anymore.

Reply:
We need `if` statement because of the `handle_updated_issue_event` that takes in two parameters, `payload` and `user_profile`